dashboard-frontend/src/helpers/study.ts

```diff
@@ -0,0 +1,25 @@
+import { StudySegmentProgressResource } from '../types/study';
+import { calculateAverageDeviation, round } from './numbers';
+
+export const computeStudyProgressDataFrom = (
+  stratumProgressList: StudySegmentProgressResource[]
+) => ({
+  currentParticipants: stratumProgressList.reduce(
+    (prev, { currentParticipants }) => prev + currentParticipants,
+    0
+  ),
+  expectedParticipants: stratumProgressList.reduce(
+    (prev, { expectedParticipants }) => prev + expectedParticipants,
+    0
+  ),
+  currentAverageDeviation: calculateAverageDeviation(
+    stratumProgressList.map(
+      ({ percentageDeviationFromGoal }) => percentageDeviationFromGoal
+    )
+  ),
+  expectedAverageDeviation: calculateAverageDeviation(
+    stratumProgressList.map(({ expectedPercentage }) =>
+      round(100 - expectedPercentage)
```

Comment:
Why is this 100 - expectedPercentage? That would imply that 100 is the desiredPercentage, no? But that's not the case.

Reply:
Yes it's a mistake, good catch 👍